modin/backends/pandas/query_compiler.py

```diff
@@ -303,7 +308,13 @@ def to_numpy(self):
     __ror__ = BinaryFunction.register(pandas.DataFrame.__ror__)
     __rxor__ = BinaryFunction.register(pandas.DataFrame.__rxor__)
     __xor__ = BinaryFunction.register(pandas.DataFrame.__xor__)
-    update = BinaryFunction.register(copy_df_for_func(pandas.DataFrame.update))
+    df_update = BinaryFunction.register(
+        copy_df_for_func(pandas.DataFrame.update), join_type="left"
```

Comment:
Why did we only implement for `join_type="left"`? The internal machinery should detect this `join_type` from the kwargs in the current implementation. Was there a bug with this?

Reply:
This `join_type` parameter is needed to compute [`joined_index`](https://github.com/modin-project/modin/blob/6637572cd415adf5aaba8a01c8f40c5d76bd2cc9/modin/engines/base/frame/data.py#L1124) correctly. Up to the current moment, value of `join_type` was `outer`. It is incorrect for `DataFrame.update` and `Series.update` operations. I think you mean the `join_type` which is in API of `DataFrame.update` and `Series.update`. Its actual name is just `join`. There is no problem with it and internally it is detected correctly for `pandas.DataFrame.update` and `pandas.Series.update` calls.

Reply:
I see, it makes sense.